designsafe/apps/api/agave/filemanager/public_search_index.py

```diff
@@ -200,53 +200,45 @@ def archive(self):
         ARCHIVE_NAME = str(self.projectId) + '_archive.zip'
         proj_dir = '/corral-repl/tacc/NHERI/published/{}'.format(self.projectId)
 
-        # open directory permissions
-        def open_perms(project_directory):
-            os.chmod('/corral-repl/tacc/NHERI/published/', 0777)
-            archive_path = os.path.join(project_directory)
-            for root, dirs, files in os.walk(archive_path):
-                os.chmod(root, 0777)
-                for d in dirs:
-                    os.chmod(os.path.join(root, d), 0777)
-                for f in files:
-                    os.chmod(os.path.join(root, f), 0777)
-
-        # close directory permissions
-        def close_perms(project_directory):
-            os.chmod('/corral-repl/tacc/NHERI/published/', 0555)
-            archive_path = os.path.join(project_directory)
-            for root, dirs, files in os.walk(archive_path):
-                os.chmod(root, 0555)
-                for d in dirs:
-                    os.chmod(os.path.join(root, d), 0555)
-                for f in files:
-                    os.chmod(os.path.join(root, f), 0555)
+        def set_perms(project_directory, octal):
+            try:
+                os.chmod('/corral-repl/tacc/NHERI/published/', octal)
+                archive_path = os.path.join(project_directory)
+                if not os.path.isdir(archive_path):
+                    raise Exception('Archive path does not exist!')
+                for root, dirs, files in os.walk(archive_path):
+                    os.chmod(root, octal)
+                    for d in dirs:
+                        os.chmod(os.path.join(root, d), octal)
+                    for f in files:
+                        os.chmod(os.path.join(root, f), octal)
+            except Exception as e:
+                logger.exception("Failed to set permissions for %s", project_directory)
```

Comment:
can you turn these into string formats?
```suggestion
                logger.exception("Failed to set permissions for {}".format(project_directory))
```

Reply:
Sure, I'll swap those out.